block-producer/src/server/api.rs

```diff
@@ -32,15 +34,32 @@ impl<T> api_server::Api for BlockProducerApi<T>
 where
     T: TransactionQueue,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(target = "miden-block-producer", skip(self, request), err)]
     async fn submit_proven_transaction(
         &self,
         request: tonic::Request<SubmitProvenTransactionRequest>,
     ) -> Result<tonic::Response<SubmitProvenTransactionResponse>, Status> {
         let request = request.into_inner();
+        debug!(target: target!(), tx = ?request.transaction);
```

Comment:
A couple of comments here:

First, we should be able to say:
```Rust
debug!(target: COMPONENT, tx = ?request.transaction);
```

Second, given that fields propagate to child spans, we should record transaction as a message. Maybe something like:
```Rust
debug!(target: COMPONENT, format!("transaction: {}", request.transaction));
```